V2_REGISTRY returns lowercase header and fails

In migrator.sh line 811 

https://github.com/docker/migrator/blob/5e7ef10ace96e34d33e8870173f230e738a292cf/migrator.sh#L811

The "if" expects "Docker-Distribution-Api-Version: registry/2.0", but it fails with my current registry 2.6.2, because the returned header is the following:
```
...
docker-distribution-api-version: registry/2.0
...
```

Am I using some wrong version of the migrator/registry or is the check maybe outdated?

Thanks! 

Lucas

Humm.... you're right, I think.

The original issue I was having was the following. I issued the following command

```
docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v /etc/docker/certs.d:/etc/docker/certs.d:ro -e V1_REGISTRY=docker.io -e V2_REGISTRY=dockerregistry.lnls-sirius.com.br -e DOCKER_HUB_ORG=lnlsdig -e V1_REPO_FILTER="epics-ioc" docker/migrator 2>&1 | tee migration.log
```

It correctly pulled all of the images but it couldn't push them to my registry:

```
[ERROR] v2 registry (dockerregistry.lnls-sirius.com.br) is not available
[!!] Verify v2 registry is functioning as expected; press any key to continue to retry [ctrl+c to abort]
```

I can reach the [dockerregistry.lnls-sirius.com.br]() 

```
nslookup dockerregistry.lnls-sirius.com.br
Server:		10.2.128.31
Address:	10.2.128.31#53

Name:	dockerregistry.lnls-sirius.com.br
Address: 10.2.128.31
```

But the command "curl -v" returns this:

```
dig@dig-cpu-test-ioc:~/Repos/migrator$ curl -v https://dockerregistry.lnls-sirius.com.br/v2

*   Trying 10.2.128.31...
* TCP_NODELAY set
* Connected to dockerregistry.lnls-sirius.com.br (10.2.128.31) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=BR; ST=Sao Paulo; L=Campinas; O=CNPEM; OU=LNLS; CN=dockerregistry.lnls-sirius.com.br
*  start date: Apr  4 20:14:05 2018 GMT
*  expire date: Apr  4 20:14:05 2019 GMT
*  common name: dockerregistry.lnls-sirius.com.br (matched)
*  issuer: C=BR; ST=Sao Paulo; L=Campinas; O=CNPEM; OU=LNLS; CN=dockerregistry.lnls-sirius.com.br
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x557d30c6adc0)
> GET /v2 HTTP/1.1
> Host: dockerregistry.lnls-sirius.com.br
> User-Agent: curl/7.52.1
> Accept: */*
> 
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
< HTTP/2 301 
< docker-distribution-api-version: registry/2.0
< location: /v2/
< content-type: text/html; charset=utf-8
< content-length: 39
< date: Tue, 22 May 2018 11:34:54 GMT
< 
<a href="/v2/">Moved Permanently</a>.

* Curl_http_done: called premature == 0
* Connection #0 to host dockerregistry.lnls-sirius.com.br left intact
```

Maybe this is related to my self-signed certificate?

--- a/migrator.sh
+++ b/migrator.sh
@@ -808,7 +808,7 @@ verify_v2_ready() {
   while [ "${V2_READY}" = "false" ]
   do
     # check to see if V2_REGISTRY is returning the expected header (see https://docs.docker.com/registry/spec/api/#api-version-check:00e71df22262087fd8ad820708997657)
-    if $(curl ${V2_OPTIONS} -Is ${V2_PROTO}://${V2_REGISTRY}/v2/ | grep ^'Docker-Distribution-Api-Version: registry/2.0' > /dev/null 2>&1)
+    if $(curl ${V2_OPTIONS} -Is ${V2_PROTO}://${V2_REGISTRY}/v2/ | grep -i ^'Docker-Distribution-Api-Version: registry/2.0' > /dev/null 2>&1)
     then
       # api version indicates v2; sets value to exit loop
       V2_READY="true"

